drivers/spi/spi_mcux_flexcomm.c

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (c) 2016, Freescale Semiconductor, Inc.
+ * Copyright (c) 2017,2019, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <drivers/spi.h>
+#include <fsl_spi.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(spi_mcux_flexcomm, CONFIG_SPI_LOG_LEVEL);
+
+#include "spi_context.h"
+
+#define SPI_CHIP_SELECT_COUNT	4
+#define SPI_MAX_DATA_WIDTH	16
+
+struct spi_mcux_config {
+	SPI_Type *base;
+	void (*irq_config_func)(struct device *dev);
+};
+
+struct spi_mcux_data {
```

Comment:
Isn't it so mcux drivers are more or less the same? Maybe at some point it would make sense to merge it in one driver (having proper ifdefs to control IP block specifics). At least when a common bug shows up, it's easier to fix in one place rather than n files.